gem/lib/textmate_grammar/grammar.rb

```diff
@@ -0,0 +1,560 @@
+                return value.transform_includes do |d|
+                    # transform includes will call this block again if d is a patternBase
+                    next d if d.is_a? PatternBase
+
+                    convert_initial_context.call(d)
+                end
+            end
+
+            return value
+        end
+        repo = repo.transform_values { |v| convert_initial_context.call(v) }
+
+        output = {
+            name: @name,
+            scopeName: @scope_name,
+        }
+
+        to_tag = lambda do |value|
+            case value
+            when Array then return {"patterns" => value.map { |v| to_tag.call(v) }}
+            when Symbol then return {"include" => "#" + value.to_s}
+            when Hash then return value
+            when String then return {"include" => value}
+            when PatternBase then return value.to_tag
+            else raise "Unexpected value: #{value.class}"
+            end
+        end
+
+        output[:repository] = repo.transform_values { |value| to_tag.call(value) }
+
+        output[:patterns] = output[:repository][:$initial_context]
+        output[:patterns] ||= []
+        output[:patterns] = output[:patterns]["patterns"] if output[:patterns].is_a? Hash
+        output[:repository].delete(:$initial_context)
+
+        output[:version] = auto_version
+        output.merge!(@keys) { |_key, old, _new| old }
+
+        @@transforms.each { |transform| output = transform.post_transform(output) }
+
+        @@linters.each do |linter|
+            raise "linting failed, see above error" unless linter.post_lint(output)
+        end
+
+        output
+    end
+
+    #
+    # Save the grammar to a path
+    #
+    # @param [Hash] options options to save_to
+    # @option options :inherit_or_embedded (:embedded) see {#generate}
+    # @option options :generate_tags [Boolean] (true) generate a list of all +:tag_as+s
+    # @option options :dir [String] the location to generate the files
+    # @option options :tag_dir [String] (File.join(options[:dir],"language_tags")) the
+    #   directory to generate language tags in
+    # @option options :syntax_dir [String] (File.join(options[:dir],"syntaxes")) the
+    #   directory to generate the syntax file in
+    # @option options :syntax_format [:json,:vscode,:plist,:textmate,:tm_language,:xml]
+    #   (:json) The format to generate the syntax file in
+    # @option options :syntax_name [String] ("#{@name}.tmLanguage") the name of the syntax
+    #   file to generate without the extension
+    #
+    # @note all keys except :dir is optional
+    # @note :dir is optional if both :tag_dir and :syntax_dir are specified
+    # @note currently :vscode is an alias for :json
+    # @note currently :textmate, :tm_language, and :xml are aliases for :plist
+    # @note later the aliased :syntax_type choices may enable compatibility features
+    #
+    # @return [void] nothing
+    #
+    def save_to(options)
+        options[:dir] ||= "."
```

Comment:
Why isn't this part of the `default` hash?

Reply:
No idea

Reply:
It is because the default syntax_dir and tag_dir are based on `options[:dir]`